### PR TITLE
Replace PoolBalanceChanged with LiquidityAdded/Removed

### LINKED ISSUE #1111: Split add / remove liquidity in 2 different events
- Deprecate `PoolBalanceChanged` in favor of `AddLiquidity` / `RemoveLiquidity` events.
- Use unsigned integers for amounts.
- Add join / exit kind

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

PoolBalanceChanged is a legacy event from v2, when the code organization was very different (e.g., join/exit code was shared). It was meant to be very generic (e.g., using int256 amounts that required interpretation).

This simplifies and clarifies liquidity operations, replacing the legacy event with two new events with unsigned amounts, and including the enumerated type.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Closes #1111

### CODE REVIEW

File: pkg/solidity-utils/contracts/helpers/CastingHelpers.sol [hunk 1]
```diff
@@ -6,20 +6,6 @@ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
...
-    function unsafeCastToInt256(
-        uint256[] memory values,
-        bool positive
-    ) internal pure returns (int256[] memory signedValues) {
-        signedValues = new int256[](values.length);
-        for (uint256 i = 0; i < values.length; ++i) {
-            signedValues[i] = positive ? int256(values[i]) : -int256(values[i]);
-        }
-    }
-
```
Comment (jubeira) [inline pkg/solidity-utils/contracts/helpers/CastingHelpers.sol:22]: :heart:

File: pkg/vault/test/.contract-sizes/Vault [hunk 2]
```diff
@@ -1,2 +1,2 @@
-Bytecode	23.563
-InitCode	24.972
\ No newline at end of file
+Bytecode	23.363
```
Comment (jubeira) [inline pkg/vault/test/.contract-sizes/Vault:1]: LOL

**Review by jubeira — APPROVED** [review]
Awesome job, thanks @EndymionJkb !

**Review by joaobrunoah — APPROVED** [review]
LGTM!

**Pull request merged**